Quartz/Engine/Source/Core/Graphics/API/GL/GLShaderPipeline.cpp

```diff
@@ -151,15 +163,36 @@ namespace
 			if (directive == "include") 
 			{
 				std::string filename = getNextToken();
+				if (filename.size() > 2)
+				{
 
-				filename.pop_back(); // remove last "
-				filename.erase(filename.begin());
-
-				std::string path = getDirname(shaderfilepath);
-				std::string filepathtolookfor = path + filename;
-
+					filename = filename.substr(1, filename.size() - 2);// remove last " and leading "
+				}
+				else
+				{
+					LFATAL("Shader source code error at %d : %d Expected filename after include!", linenum, colnum);
+					assert(false);
+				}
+				std::string filepathtolookfor = dirpath + filename;
 				std::string contentsToPaste = qz::utils::FileIO::readAllFile(filepathtolookfor);
-				(*currentStage)->source().append(contentsToPaste);
+				if (contentsToPaste.size() > 0)
+				{
+					//handles even if source if null
+					if ((*currentStage))
```

Comment:
I'd remove the extra pair of brackets here
```suggestion
					if (*currentStage)
```